scene/resources/animation.cpp

```diff
@@ -5947,17 +5959,21 @@ Variant Animation::blend_variant(const Variant &a, const Variant &b, float c) {
 }
 
 Variant Animation::interpolate_variant(const Variant &a, const Variant &b, float c, bool p_snap_array_element) {
-	if (a.get_type() != b.get_type() && !a.is_array()) {
-		return a;
+	if (a.get_type() != b.get_type()) {
+		if (a.is_num() && b.is_num()) {
+			return interpolate_variant(cast_to_blendwise(a), cast_to_blendwise(b), c);
```

Comment:
```suggestion
			return interpolate_variant(cast_to_blendwise(a), cast_to_blendwise(b), c, p_snap_array_element);
```
Missed argument

Reply:
Good catch, although at this point we already know that this argument is not going to be used (we've checked that we're dealing with numbers, not arrays), so we can either keep it for consistency or drop it for conciseness. I had the argument there in my earlier commits but then noticed that it isn't passed down in any of the other recursive calls in the method, so I thought I'd follow that lead.

Reply:
Good point 🙂